### ISSUE TITLE
Enquiry

### ISSUE THREAD

**Issue by Cownic** [aryanitish/thesis#1]
Hi, I a undergraduate student working on Gem5, learning how to use the PARSEC benchmark, could i ask you some questions related to it

**Comment by aryanitish** [comment]
Hello there,
Yes ofcourse, I would be happy to help.

**Comment by Cownic** [comment]
ohhh think you so much for being willing to help. 
it was stated in your readme that it uses Linux in addition to the parsec image. Do I have to download it from somewhere and where do I put it ? 
I don't see it in the directory layout that U have on the readme.

**Comment by aryanitish** [comment]
Sure, let me clarify it for you.
The repository is a copy of [this](https://github.com/gem5/gem5-resources/tree/stable/src/parsec), so the README is also the same. I would recommend to follow this link instead of my repository to run simulations.
I implemented changes for my thesis in some config files which is described in [this file](https://github.com/aryanitish/thesis/blob/master/readme_nitish.txt).
Regarding your doubt, there are two things which a full system gem5 simulation needs: a linux kernel and a disk-image(in which parsec is already installed). The kernel can be obtained from [kernel](https://github.com/gem5/gem5-resources/tree/stable/src/linux-kernel) and the disk-image will be created once you go through the steps in the README. 
For starting out, it is better to follow the official gem5 repository and to obtain resources such as kernel and disk-images follow [this](https://github.com/gem5/gem5-resources/tree/stable?tab=readme-ov-file#resource-parsec-benchmark-suite)

**Comment by Cownic** [comment]
Thank you for the comprehensive reply. 
I am grateful for the links that you have provided.
For the kernel stuff, do i place it under the directory of parsec

**Comment by Cownic** [comment]
Thank you for the comprehensive reply.
I am grateful for the links that you have provided.
For the kernel stuff, do i place it under the directory of parsec 
parsec/
  |___ gem5/                                   # gem5 folder
  |
  |___ disk-image/
  |      |___ build.sh                         # the script downloading packer binary and building the disk image
  |      |___ shared/
  |      |___ parsec/
  |             |___ parsec-image/
  |             |      |___ parsec             # the disk image will be here
  |             |___ parsec.json               # the Packer script
  |             |___ parsec-install.sh         # the script to install PARSEC
  |             |___ post-installation.sh      # the script to install m5
  |             |___ runscript.sh              # script to run each workload
  |             |___ parsec-benchmark          # the parsec benchmark suite
  |_______linux ?
  |
  |___ README.md

**Comment by aryanitish** [comment]
Actually you do not need to download anything. The script `x86-parsec-benchmarks.py ` automatically downloads the kernel and disk-image. 
Build gem5 first and then run the last command in the readme with your preferred benchmark and size:
```shell
# In the gem5 directory
build/X86/gem5.opt \
configs/example/gem5_library/x86-parsec-benchmarks.py \
--benchmark <benchmark_program> \
--size <size> 
```

**Comment by Cownic** [comment]
ohhhh alright that works for me Thank you for the guidance.
One last question, the x86-parsec-benchmarks.py uses MESI_Two_Level as the cache protocol, do u know if its possible to change it to the other cache protocols that are supported by gem5, such as MI, MOESI etc ?

**Comment by aryanitish** [comment]
yes you can use any of the gem5 supported cache protocols. If you see in the x86-parsec-benchmarks.py script at given lines [70](https://github.com/aryanitish/thesis/blob/6a16ee3f80812717848dd566324134872047762f/gem5/configs/example/gem5_library/x86-parsec-benchmarks.py#L70), [122](https://github.com/aryanitish/thesis/blob/6a16ee3f80812717848dd566324134872047762f/gem5/configs/example/gem5_library/x86-parsec-benchmarks.py#L122) you can understand how it imports the already built components. Then you can modify these lines to use other cache heirarchies/protocols. Not all the supported coherence protocols are prebuilt as a component but you can learn to make your own with the gem5-ruby tutorials. I hope it helps
